### PR TITLE
bug 1406546: refactor groovy code to handle kumascript as well as kuma

### PR DESCRIPTION
**Pull request by escattone**
This PR refactors [utils.groovy](https://github.com/mozilla/kuma/blob/master/Jenkinsfiles/utils.groovy) to handle kumascript as well.

### CODE REVIEW

**Comment by escattone** [comment]
I'd like to tell [the final `junit` step](https://github.com/mozilla/kuma/blob/7a7120f12c9349415877eacb040bfb393099d6d1/Jenkinsfile#L53) to ignore the case when there are no results, but I don't know how to do that. From my research, I think there's a setting on the JUnit plugin we need to enable, but I don't see that we have the JUnit plugin installed. Does anyone more familiar with Jenkins know how to do this?

File: Jenkinsfiles/utils.groovy [hunk 1]
```diff
@@ -2,10 +2,38 @@
  * Define utility functions.
  */
 
-def checkout_github(repo, branch, relative_target_dir) {
+def get_commit_tag() {
+    return env.GIT_COMMIT.take(7)
```
Comment (metadave) [inline Jenkinsfiles/utils.groovy]: whats does the value 7 correspond to?

File: Jenkinsfiles/utils.groovy [hunk 2]
```diff
@@ -2,10 +2,38 @@
...
+    if (env.BRANCH_NAME.startsWith('stage-')) {
+        return 'stage'
+    }
+    throw new Exception(
+        'Unable to determine the target from the branch name.'
+    )
+}
+
+def get_repo_name() {
+    if (env.JOB_NAME.startsWith('mdn_multibranch')) {
```
Comment (metadave) [inline Jenkinsfiles/utils.groovy]: it might be nice to see both branch names pulled out into constants

File: Jenkinsfiles/utils.groovy (hunk 1 above)
Comment (escattone) [reply Jenkinsfiles/utils.groovy]: It means "take the first 7 characters" of the git commit hash (so we match the tag we use for quay.io).

Comment (metadave) [reply Jenkinsfiles/utils.groovy]: ah, ok, I'm not used to the Groovy APIs.

**Comment by escattone** [comment]
Thanks for noticing that I should be using `github.com/mozmeao` instead of `github.com/mozmar` in https://github.com/mdn/kumascript/pull/395. I updated it here as well.

**Comment by escattone** [comment]
It occurred to me that it may be wise to ensure that the Docker image can be pulled (that it exists on quay.io). If the image failed to be registered on quay.io, then the push will "succeed", but the pods will spin in ImagePullBackOff.

**Pull request merged**